server/player/player.go

```diff
@@ -1147,6 +1187,75 @@ func (p *Player) Jump() {
 	}
 }
 
+// Sleep makes the player sleep at the given position. If the position does not map to a bed (specifically the head side),
+// the player will not sleep.
+func (p *Player) Sleep(pos cube.Pos) {
+	ctx, sendReminder := event.C(p), true
+	if p.Handler().HandleSleep(ctx, &sendReminder); ctx.Cancelled() {
+
+		return
+	}
+	tx := p.tx
+
+	if b, ok := tx.Block(pos).(block.Bed); ok {
+		if b.Sleeper != nil {
+			// The player cannot sleep here.
+			return
+		}
```

Comment:
```suggestion
```

This is the responsibility of `block.Bed`. Exiting early here is also awkard because the sleep event has already been emitted by this point.